### PR TITLE
MGMT-16515: in order to set machine network in node ip hint file in post pivot we should take it from target

### PR DESCRIPTION
**Pull request by tsorya**
[MGMT-16515](https://issues.redhat.com//browse/MGMT-16515): in order to set machine network in node ip hint file in post pivot we should take it from target

# Background / Context
[MGMT-16515](https://issues.redhat.com//browse/MGMT-16515): in order to set machine network in node ip hint file in post pivot we should take it from target

# Issue / Requirement / Reason for change

# Solution / Feature Overview
This pr adds logic to take machine network from install-config of upgraded sno and set it in seed reconfiguration api.
Post pivot will use this field and set it in nodeip hint file in order require from node-configuration service to set kubelet ip from the machine network we provided

# Implementation Details

# Other Information

### CODE REVIEW

File: utils/client_helper.go [hunk 1]
```diff
@@ -145,6 +153,28 @@ func getNodeInternalIP(node corev1.Node) (string, error) {
...
+		if _, _, err := net.ParseCIDR(mn.CIDR); err != nil {
+			return "", fmt.Errorf("MachineNetwork has an invalid CIDR: %s", mn.CIDR)
+		}
+		if _, ipnet, err := net.ParseCIDR(mn.CIDR); err == nil && ipnet.Contains(net.ParseIP(nodeIp)) {
+			return mn.CIDR, nil
+		}
+	}
+
+	// retuning empty string if no match found in order to be backward compatible as there is a possibility that
+	// that we upgrade cluster that was previously upgraded with lc-agent version that didn't set new machineNetwork
```
Comment (eranco74) [inline utils/client_helper.go:172]:
```suggestion
	// that we upgrade cluster that was previously upgraded with lca-agent version that didn't set new machineNetwork
```

Comment (eranco74) [inline utils/client_helper.go:172]: Add logging, with the reason we return empty (node IP doesn't match?, no machine network?, machine network is invalid...)

**Review by eranco74 — COMMENTED** [review]
Will this play nice with https://github.com/openshift-kni/lifecycle-agent/pull/422?

**Comment by tsorya** [comment]
> Will this play nice with #422?

this is exactly the idea

File: utils/client_helper.go [hunk 2]
```diff
@@ -145,6 +156,31 @@ func getNodeInternalIP(node corev1.Node) (string, error) {
...
+	for _, mn := range installConfig.Networking.MachineNetwork {
+		if _, _, err := net.ParseCIDR(mn.CIDR); err != nil {
+			return "", fmt.Errorf("machineNetwork has an invalid CIDR: %s", mn.CIDR)
+		}
+		if _, ipnet, err := net.ParseCIDR(mn.CIDR); err == nil && ipnet.Contains(net.ParseIP(nodeIp)) {
+			return mn.CIDR, nil
+		}
+	}
+
+	log.Warnf("failed to find machine network in <%s> for node ip: %s. Returning empty string",
```
Comment (eranco74) [inline utils/client_helper.go:174]: The order is wrong

**Review by eranco74 — COMMENTED** [review]
Just a nit

File: utils/client_helper.go (hunk 2 above)
Comment (tsorya) [reply utils/client_helper.go:174]: oops

**Review by eranco74 — COMMENTED** [review]
/lgtm

**Pull request merged**